extensions/amp-social-share/1.0/social-share.js

```diff
@@ -74,12 +65,55 @@ export function SocialShare(props) {
       onClick={() => handleActivation(finalEndpoint, checkedTarget)}
       style={{...size, ...props['style']}}
     >
+      {processChildren(props, size)}
+    </div>
+  );
+}
+
+/**
+ * If the specified type can have children and children exist return children.
+ * Otherwise return the default icon for the specified type.  If the specified
+ * type can have its color / background changed, set those to color /
+ * background specified in props, otherwise use defaults, or do not set if
+ * defaults do not exist.
```

Comment:
nit: Can this be more clear? Something like:
```
Computes children based on four possible configurations in order:
  - Children provided
  - Color provided
  - Background color provided
  - Type provided

Otherwise, use icon defaults for all pre-configured types and empty <svg> otherwise.
```

Btw, I assume `backgroundColor` can apply and `color` is noop if provided for a non-configured type. Is that right?

Reply:
Thanks, I have updated the comments to be a bit more clear.  For non-configured type, both `color` and `backgroundColor` can apply.  Although there won't really be anything for `color` to apply to unless the publisher provides children as well.